### PR TITLE
Mod/pipe 509 reduce load transactions to delta django command

### PR DESCRIPTION
**Pull request by cthomas-frb**
**Description:**
High level description of what the PR addresses should be put here. Should be detailed enough to communicate to a PO what this PR addresses without diving into the technical nuances

**Technical details:**
The technical details for the knowledge of other developers. Any detailed caveats or specific deployment steps should be outlined here.

**Requirements for PR merge:**

1. [ ] Unit & integration tests updated
2. [ ] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [ ] Matview impact assessment completed
5. [ ] Frontend impact assessment completed
6. [ ] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [ ] Jira Ticket [DEV-123](https://federal-spending-transparency.atlassian.net/browse/DEV-123):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/etl/management/commands/load_transactions_in_delta.py [hunk 1]
```diff
@@ -225,10 +226,15 @@ def award_id_lookup_pre_delete(self):
         #       returned by a SELECT subquery inside the 'IN').  Thus, this code should return a dataframe directly,
         #       create a temporary view from the dataframe in award_id_lookup_post_delete, and use that temporary
         #       view to either do a subquery in the 'IN' clause or to JOIN against.
-        possibly_modified_award_ids = [str(row["award_id"]) for row in self.spark.sql(sql).collect()]
+        if table_exists:
```
Comment (dporth-frb) [inline usaspending_api/etl/management/commands/load_transactions_in_delta.py:229]: You can reduce the complexity of this by moving the contents inside the if block out and changing the predicate to if not table_exists...raise exception.

```
if not table_exists:
     raise Exception(f"Table: int.{self.etl_level} does not exist.")
 possibly_modified_award_ids = [str(row["award_id"]) for row in self.spark.sql(sql).collect()]
```

**Review by dporth-frb — CHANGES_REQUESTED** [review]

**Review by dporth-frb — CHANGES_REQUESTED** [review]

File: usaspending_api/etl/tests/integration/test_load_transactions_in_delta.py [hunk 2]
```diff
@@ -931,27 +925,16 @@ def _happy_path_test_core(
...
         expected_transaction_id_lookup_pops,
     ):
-        # 1. Test calling load_transactions_in_delta with the etl-level set to the proper sequencing of
-        # initial_run, then transaction_id_lookup
-
+        # Call load_transactions_in_delta with the etl-level set to the proper sequencing of initial_run
         # Since these tests only care about the condition of the transaction_id_lookup table after various
         # operations, only load the essential tables to Delta, and don't copy the raw transaction tables to int.
         TestInitialRun.initial_run(s3_data_bucket, load_other_raw_tables=load_other_raw_tables, initial_copy=False)
-        call_command("load_transactions_in_delta", "--etl-level", "transaction_id_lookup")
```
Comment (dporth-frb) [inline usaspending_api/etl/tests/integration/test_load_transactions_in_delta.py:940]: Why did you remove this and the verify below?

File: usaspending_api/etl/management/commands/load_transactions_in_delta.py (hunk 1 above)
Comment (cthomas-frb) [reply usaspending_api/etl/management/commands/load_transactions_in_delta.py]: Done.

File: usaspending_api/etl/management/commands/load_transactions_in_delta.py [hunk 3]
```diff
@@ -287,15 +293,18 @@ def delete_records_sql(self):
...
-        sql = f"""
-            MERGE INTO int.{self.etl_level}
-            USING (
-                {subquery}
-            ) AS deleted_records
-            ON {self.etl_level}.{id_col} = deleted_records.id_to_remove
-            WHEN MATCHED
-            THEN DELETE
-        """
+        if table_exists:
```
[the remark this answers is not in the document]
Comment (cthomas-frb) [reply usaspending_api/etl/management/commands/load_transactions_in_delta.py]: Done.

File: usaspending_api/etl/tests/integration/test_load_transactions_in_delta.py (hunk 2 above)
Comment (cthomas-frb) [reply usaspending_api/etl/tests/integration/test_load_transactions_in_delta.py:940]: It was removed as part of combining the tests. The `verify` was moved lower in the test and there is a `call_command` lower as well. Before this change the first test would load up the mocked data and check that it was loaded correctly. Then there were 3 tests after that for delete/insert and updating the data. Those 4 original tests have been combined into 2 tests now. The first one loads data and does one delete/insert then checks everything with the `verify`. The second test does a more advanced delete/insert and checks certain edge cases are taken care of.
